packages/core/src/lib/connector.ts

```diff
@@ -0,0 +1,20 @@
+import pRetry from 'p-retry';
+
+import { hasConnectorWithId } from '@/queries/connector';
+import { buildIdGenerator } from '@/utils/id';
+
+const connectorId = buildIdGenerator(10);
+
+export const generateConnectorId = async (retries = 500) =>
+  pRetry(
+    async () => {
+      const id = connectorId();
+
+      if (!(await hasConnectorWithId(id))) {
+        return id;
+      }
+
+      throw new Error('Cannot generate connector ID in reasonable retries');
+    },
+    { retries, factor: 0 } // No need for exponential backoff
+  );
```

Comment:
nit: looks like the same as `generateUserId` could extract using the same util method?  pass the findExistingId method as a param?

Reply:
Actually we can simply use `nanoid()`, `user` is a special case, you can refer to other tables.

Reply:
seems that there is no need to retry?